crates/sui-indexer/src/handlers/checkpoint_handler.rs

```diff
@@ -160,22 +158,14 @@ where
         &self,
         seq: CheckpointSequenceNumber,
     ) -> Result<CheckpointData, IndexerError> {
-        let mut checkpoint = self
-            .rpc_client
-            .read_api()
-            .get_checkpoint(seq.into())
-            .await
-            .map_err(|e| {
-                IndexerError::FullNodeReadingError(format!(
-                    "Failed to get checkpoint with sequence number {} and error {:?}",
-                    seq, e
-                ))
-            });
-        // this happens very often b/c checkpoint indexing is faster than checkpoint
-        // generation. Ideally we will want to differentiate between a real error and
-        // a checkpoint not generated yet.
-        while checkpoint.is_err() {
-            tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+        let mut checkpoint;
+        loop {
+            // sleep for 0.1 second and retry if latest checkpoint is not available yet
+            tokio::time::sleep(std::time::Duration::from_millis(100)).await;
```

Comment:
this will generate 10 times more traffic to the FN.... do we really want this?

Reply:
good point, reverting this now